pkg/skaffold/lsp/util.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lsp
+
+import (
+	"context"
+	"net/url"
+	"strings"
+
+	"go.lsp.dev/uri"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/output/log"
+)
+
+func uriToFilename(v uri.URI) string {
+	s := string(v)
+	fixed, ok := fixURI(s)
+
+	if !ok {
+		unescaped, err := url.PathUnescape(s)
+		if err != nil {
+			log.Entry(context.TODO()).Warnf("Unsupported URI (not a filepath): %q\n", s)
+		} else {
+			log.Entry(context.TODO()).Warnf("Unsupported URI (not a filepath): %q\n", unescaped)
+		}
+		return ""
+	}
+	v = uri.URI(fixed)
+
+	return v.Filename()
+}
+
+// workaround for unsupported file paths (git + invalid file://-prefix )
+func fixURI(s string) (string, bool) {
+	if strings.HasPrefix(s, "git:/") {
+		return "file:///" + s[5:], true
```

Comment:
nit: make this consistent with the case below
```suggestion
		return "file:///" + s[len("git:/"):], true
```

Might be better to even put these strings in variables and use them in both the `strings.HasPrefix()` call and when getting the range of the slice

Reply:
done